plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeSystemTablesWithAccessTracking.java

```diff
@@ -0,0 +1,240 @@
+            assertUpdate("INSERT INTO test_simple_table_pushdown_version_filter VALUES 3", 1);
+            assertUpdate("INSERT INTO test_simple_table_pushdown_version_filter VALUES 4", 1);
+            assertQuerySucceeds("ALTER TABLE test_simple_table_pushdown_version_filter EXECUTE OPTIMIZE");
+
+            fileSystemFactory.resetOpenCount();
+            // equal
+            assertThat(query("SELECT version, operation FROM \"test_simple_table_pushdown_version_filter$history\" WHERE version = 0 AND operation = 'CREATE TABLE'"))
+                    .matches("VALUES (BIGINT '0', VARCHAR 'CREATE TABLE')");
+            assertThat(fileSystemFactory.getOpenCount()).containsExactlyInAnyOrderEntriesOf(
+                    ImmutableMap.of(
+                            "00000000000000000000.json", 1,
+                            "00000000000000000005.checkpoint.parquet", 1,
+                            "00000000000000000006.json", 16,
+                            "_last_checkpoint", 17));
+            fileSystemFactory.resetOpenCount();
+            // less than
+            assertThat(query("SELECT version, operation FROM \"test_simple_table_pushdown_version_filter$history\" WHERE version < 1"))
+                    .matches("VALUES (BIGINT '0', VARCHAR 'CREATE TABLE')");
+            assertThat(fileSystemFactory.getOpenCount()).containsExactlyInAnyOrderEntriesOf(
+                    ImmutableMap.of(
+                            "00000000000000000000.json", 1,
+                            "00000000000000000006.json", 16,
+                            "_last_checkpoint", 17));
+            fileSystemFactory.resetOpenCount();
+            // less than value greater than the number of transactions
+            assertThat(query("SELECT version, operation FROM \"test_simple_table_pushdown_version_filter$history\" WHERE version < 10"))
+                    .matches("""
+                            VALUES
+                                (BIGINT '0', VARCHAR 'CREATE TABLE'),
+                                (BIGINT '1', VARCHAR 'WRITE'),
+                                (BIGINT '2', VARCHAR 'WRITE'),
+                                (BIGINT '3', VARCHAR 'WRITE'),
+                                (BIGINT '4', VARCHAR 'WRITE'),
+                                (BIGINT '5', VARCHAR 'OPTIMIZE')
+                            """);
+            assertThat(fileSystemFactory.getOpenCount()).containsExactlyInAnyOrderEntriesOf(
+                    ImmutableMap.of(
+                            "00000000000000000000.json", 1,
+                            "00000000000000000001.json", 1,
+                            "00000000000000000002.json", 1,
+                            "00000000000000000003.json", 1,
+                            "00000000000000000004.json", 1,
+                            "00000000000000000005.json", 1,
+                            "00000000000000000006.json", 17,
+                            "_last_checkpoint", 17));
```

Comment:
Do you have a sense of why these two are so high on these queries?

Reply:
I counted while debugging `9` times the call to `io.trino.connector.system.SystemTablesMetadata#checkAndGetTable` while performing a `SELECT` from the `$history` system metadata table.

Each of this calls eventually causes a subsequent `DeltaLakeMetadata#getTableHandle()` call which calls `TransactionLogParser#tryReadLastCheckpoint`.

I was planning to create an issue to further investigate whether we could spare some `getTableHandle()` calls.